packages/core/core/src/Graph.js

```diff
@@ -115,17 +115,17 @@ export default class Graph<TNode: Node> {
   removeNode(node: TNode): this {
     let removed = new this.constructor();
 
+    for (let from of this.inboundEdges.get(node.id)) {
```

Comment:
yeah this makes a lot more sense. we should probably move up removing outbound edges too, since those are also edges to a node that's no longer in the graph.